RecoEgamma/ElectronIdentification/src/ElectronMVAEstimatorRun2Fall17.cc

```diff
@@ -0,0 +1,428 @@
+  float PreShowerOverRaw  = superCluster->preshowerEnergy() / superCluster->rawEnergy();
+
+  // To get to CTF track information in pat::Electron, we have to have the pointer
+  // to pat::Electron, it is not accessible from the pointer to reco::GsfElectron.
+  // This behavior is reported and is expected to change in the future (post-7.4.5 some time).
+  bool validKF= false;
+  reco::TrackRef myTrackRef = eleRecoPtr->closestCtfTrackRef();
+  const pat::Electron * elePatPtr = dynamic_cast<const pat::Electron *>(eleRecoPtr);
+  // Check if this is really a pat::Electron, and if yes, get the track ref from this new
+  // pointer instead
+  if( elePatPtr != nullptr )
+    myTrackRef = elePatPtr->closestCtfTrackRef();
+  validKF = (myTrackRef.isAvailable() && (myTrackRef.isNonnull()) );
+
+  //Pure tracking variables
+  float kfhits         = (validKF) ? myTrackRef->hitPattern().trackerLayersWithMeasurement() : -1. ;
+  float kfchi2          = (validKF) ? myTrackRef->normalizedChi2() : 0;
+  float gsfchi2         = eleRecoPtr->gsfTrack()->normalizedChi2();
+
+  // Energy matching
+  float fbrem           = eleRecoPtr->fbrem();
+
+  float gsfhits         = eleRecoPtr->gsfTrack()->hitPattern().trackerLayersWithMeasurement();
+  float expectedMissingInnerHits = eleRecoPtr->gsfTrack()
+    ->hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS);
+
+  reco::ConversionRef conv_ref = ConversionTools::matchedConversion(*eleRecoPtr,
+                                    conversions,
+                                    theBeamSpot->position());
+  float convVtxFitProbability = -1.;
+  if(!conv_ref.isNull()) {
+    const reco::Vertex &vtx = conv_ref.get()->conversionVertex(); if (vtx.isValid()) {
+      convVtxFitProbability = (float)TMath::Prob( vtx.chi2(), vtx.ndof());
+    }
+  }
+
+  float EoP             = eleRecoPtr->eSuperClusterOverP();
+  float eleEoPout       = eleRecoPtr->eEleClusterOverPout();
+  float pAtVertex            = eleRecoPtr->trackMomentumAtVtx().R();
+  float IoEmIoP         = (1.0/eleRecoPtr->ecalEnergy()) - (1.0 / pAtVertex );
+
+  // Geometrical matchings
+  float deta            = eleRecoPtr->deltaEtaSuperClusterTrackAtVtx();
+  float dphi            = eleRecoPtr->deltaPhiSuperClusterTrackAtVtx();
+  float detacalo        = eleRecoPtr->deltaEtaSeedClusterTrackAtCalo();
+
+  if(withIso)
+  {
+
+    // Isolation variables
+    float ele_pfChargedHadIso   = (eleRecoPtr->pfIsolationVariables()).sumChargedHadronPt ; //chargedHadronIso();
+    float ele_pfNeutralHadIso   = (eleRecoPtr->pfIsolationVariables()).sumNeutralHadronEt ; //neutralHadronIso();
+    float ele_pfPhotonIso       = (eleRecoPtr->pfIsolationVariables()).sumPhotonEt; //photonIso();
+
+    /*
+     * Packing variables for the MVA evaluation.
+     * CAREFUL: It is critical that all the variables that are packed into “vars” are
+     * exactly in the order they are found in the weight files
+     */
+    std::vector<float> vars = std::move( packMVAVariables(
```

Comment:
the std::move() is unnecessary its already an rvalue. Regardless, having looked at packMVAVariables (it just makes a vector) it would be better to just do
std::vector<float> vars {   
instead as it would achieve exactly the same thing and is  much more readable